pages/workspace/workspace.js

```diff
@@ -22,6 +22,9 @@ router.get('/:workspace_id/:action', asyncHandler(async (req, res, next) => {
         const state = 'stopped';
         await workspace.updateState(workspace_id, state);
         res.redirect(`/workspace/${workspace_id}`);
+    } else if (action === 'grade') {
+        await workspace.controlContainer(workspace_id, 'grade');
+        res.redirect(`/workspace/${workspace_id}`);     // refresh anyway
```

Comment:
We don't need an action here --- the `grade/` endpoint will only be used by the PL servers themselves during the grading process.